Apps/System/music.c

```diff
@@ -0,0 +1,461 @@
+        // Store time for status bar
+        memcpy(&s_last_time, tick_time, sizeof(struct tm));
+        layer_mark_dirty(s_main_layer);
+    }
+}
+
+static void implementation_record_setup(Animation *animation) {
+    s_animating_disk_change = true;
+    // TODO set next_disk_color based on title + artist,
+    // to let each track have the same color between sessions
+    // Avoid color black and white (Changing the stroke color would be an alternative)
+    s_next_disk_color =  (GColor8) { .argb = ((rand() % 0b00111111) + 0b11000000)  };
+
+    // Set the initial positions for our two animated records
+    if(s_skip_value > 0){
+        s_next_disk_pos = GPoint(RECORD_CENTER_X + RECORD_SWOOSH_DISTANCE, RECORD_CENTER_Y);
+    } else {
+        s_next_disk_pos = GPoint(RECORD_CENTER_X - RECORD_SWOOSH_DISTANCE, RECORD_CENTER_Y);
+    }
+}
+
+static void implementation_record_update(Animation *animation,
+                                  const AnimationProgress distance_normalized) {
+    if(s_skip_value > 0) {
+        s_curr_disk_pos = GPoint(LERP(s_curr_disk_pos.x, RECORD_CENTER_X - RECORD_SWOOSH_DISTANCE), RECORD_CENTER_Y);
+    } else {
+        s_curr_disk_pos = GPoint(LERP(s_curr_disk_pos.x, RECORD_CENTER_X + RECORD_SWOOSH_DISTANCE), RECORD_CENTER_Y);
+    }
+    s_next_disk_pos =     GPoint(LERP(s_next_disk_pos.x, RECORD_CENTER_X),                          RECORD_CENTER_Y);
+    layer_mark_dirty(s_main_layer);
+}
+
+static void implementation_record_teardown(Animation *animation) {
+    // Now that the animation is done, our next disk becomes the current one
+    s_curr_disk_color = s_next_disk_color;
+    s_curr_disk_pos = s_next_disk_pos;
+    // We successfully skipped, reduce the skip value
+    if (s_skip_value > 0) {
+        s_skip_value--;
+    } else if (s_skip_value < 0) {
+        s_skip_value++;
+    }
+    if(s_skip_value != 0) {
+        // There's more skipping to be done, just leave the arm and continue swapping
+        animation_record();
+    } else {   
+        // We are done, move the arm to the start position
+        s_animating_disk_change = false;
+        if(!s_was_paused) {
+            s_is_paused = false;
+            s_animating_arm_change = true;
+            animation_arm(ARM_START_ANGLE, ARM_QUICK_SPEED);
+        }
+    }
+}
+
+static const AnimationImplementation implementation_record = {
+    .setup = implementation_record_setup,
+    .update = implementation_record_update,
+    .teardown = implementation_record_teardown
+};
+
+static void animation_record() {
+    animation_schedule(s_animation_record_ptr);
+}
+
+static void implementation_arm_setup(Animation *animation) {
+    s_animation_start_arm_angle = s_arm_angle;
+    s_animating_arm_change = true;
+}
+
+static void implementation_arm_update(Animation *animation,
+                                      const AnimationProgress distance_normalized) {
+    s_arm_angle = LERP(s_animation_start_arm_angle, s_animation_end_arm_angle);
```

Comment:
I think at some point we'll end up pulling LERP out of the header -- it probably shouldn't be there, anyway.  So if you want a LERP macro, probably defining it locally is the way to go.